### PR TITLE
[MAINT] Format nilearn/plotting: apply isort and fix flake8 errors

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to https://github.com/nilearn/nilearn/issues/2528

Changes proposed in this pull request:

- [x] run isort
- [x] run pyupgrade
- [x] run flynt
- [x] fix remaining flake8 issues
- [x] update flake8 / isort CI

### CODE REVIEW

File: nilearn/conftest.py
Comment (Remi-Gau) [inline nilearn/conftest.py:1]: Extracted some of the fixtures in here to allow removing some `noqa`.

I suspect that this should be further refactored in another PR in the future when we try to improve the use of fixtures more generally in the test suite

Comment (ymzayek) [reply nilearn/conftest.py:1]: Yes that would be great!

File: nilearn/plotting/html_stat_map.py [hunk 1]
```diff
@@ -315,7 +318,7 @@ def _json_view_size(params):
 
 
 def _get_bg_mask_and_cmap(bg_img, black_bg):
-    """Helper function of _json_view_data."""
+    """Help of _json_view_data."""
```
Comment (ymzayek) [inline nilearn/plotting/html_stat_map.py:321]:
```suggestion
    """Get background data for _json_view_data."""
```

File: nilearn/plotting/img_plotting.py [hunk 2]
```diff
@@ -417,8 +415,9 @@ def plot_anat(anat_img=MNI152TEMPLATE, cut_coords=None,
               draw_cross=True, black_bg='auto', dim='auto', cmap=plt.cm.gray,
               colorbar=False, cbar_tick_format="%.2g", vmin=None,
               vmax=None, **kwargs):
-    """Plot cuts of an anatomical image (by default 3 cuts:
-    Frontal, Axial, and Lateral)
+    """Plot cuts of an anatomical image.
+
+    Bby default 3 cuts: Frontal, Axial, and Lateral.
```
Comment (ymzayek) [inline nilearn/plotting/img_plotting.py:420]:
```suggestion
    By default 3 cuts: Frontal, Axial, and Lateral.
```

File: nilearn/plotting/surf_plotting.py [hunk 3]
```diff
@@ -570,7 +558,7 @@ def plot_surf(surf_mesh, surf_map=None, bg_map=None,
               cbar_vmin=None, cbar_vmax=None, cbar_tick_format="auto",
               title=None, title_font_size=18, output_file=None, axes=None,
               figure=None, **kwargs):
-    """Plotting of surfaces with optional background and data
+    """Plot of surfaces with optional background and data.
```
Comment (ymzayek) [inline nilearn/plotting/surf_plotting.py:561]:
```suggestion
    """Plot surfaces with optional background and data.
```

**Review by ymzayek — COMMENTED** [review]
Minor comments. I haven't reviewed the tests yet.

File: nilearn/plotting/img_plotting.py [hunk 4]
```diff
@@ -1011,7 +1015,7 @@ def plot_glass_brain(stat_map_img,
         absolute value will be used (rendering positive and negative
         values in the same manner). If set to false the sign of the
         maximum intensity will be represented with different colors.
-        See http://nilearn.github.io/auto_examples/01_plotting/plot_demo_glass_brain_extensive.html
+        See http://nilearn.github.io/auto_examples/01_plotting/plot_demo_glass_brain_extensive.html # noqa
```
Comment (ymzayek) [inline nilearn/plotting/img_plotting.py:1018]:
```suggestion
         See
        :ref:`sphx_glr_auto_examples_01_plotting_plot_demo_glass_brain_extensive.py` # noqa
```

File: nilearn/plotting/cm.py [hunk 5]
```diff
@@ -110,48 +102,47 @@ def alpha_cmap(color, name='', alpha_min=0.5, alpha_max=1.):
...
+        f'{name}_transparent', cmapspec, _rcParams['image.lut'])
     cmap._init()
     cmap._lut[:, -1] = _np.linspace(alpha_min, alpha_max, cmap._lut.shape[0])
     cmap._lut[-1, -1] = 0
     return cmap
 
 
-
-################################################################################
+###############################################################################
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/plotting/cm.py:114]: I would say that we either have them everywhere and we agree on a typical way to organize each module or we have them nowhere.

Since the latter is the norm in the codebase, I will remove them but I would not be against a soft guidelines to help organize function / classes in each file.

[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/plotting/cm.py:109]: OK I could not get myself to remote those headings as they do help read / navigate the code.

File: nilearn/plotting/cm.py [hunk 6]
```diff
@@ -110,48 +102,47 @@ def alpha_cmap(color, name='', alpha_min=0.5, alpha_max=1.):
...
+                             swap_order=('red', 'blue', 'green')),
+    black_red=_cm.hot._segmentdata.copy(),
 )
 
 if hasattr(_cm, 'ocean'):
     # MPL 0.99 doesn't have Ocean
-    _cmaps_data['ocean_hot'] =  _concat_cmap(_cm.ocean, _cm.hot_r)
-if hasattr(_cm, 'afmhot'): # or afmhot
+    _cmaps_data['ocean_hot'] = _concat_cmap(_cm.ocean, _cm.hot_r)
+if hasattr(_cm, 'afmhot') :  # or afmhot
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/plotting/cm.py]: pulled all the comments above those 2 if blocks

**Comment by Remi-Gau** [comment]
draft until #3635 is in

File: nilearn/conftest.py
Comment (ymzayek) [inline nilearn/conftest.py:1]: Just a couple of thoughts:

- The new plotting fixtures seem to be specific to the plotting module while the other fixtures are a bit more general, perhaps they can stay in the plotting module
- Can any "create fake data" function from _utils.data_gen be reused to create these fixtures? Or we can specify new functions in data_gen and then specify the fixtures here (or under plotting) if we see that they can be useful in other modules/fixtures

These are maybe for another PR but I also think that moving the fixtures would be better done under a refactoring PR

Comment (ymzayek) [reply nilearn/conftest.py:1]: BTW to add locations of fixtures for pytest to find you can add the following code to `conftest.py`:

```python
pytest_plugins = [
    "nilearn.plotting.tests.test_img_plotting.testing_utils",
]
```

And then change the names of the fixtures and then you don't have to undo the changes in the files that use those fixtures

Comment (Remi-Gau) [reply nilearn/conftest.py:1]: I only moved the fixtures. I started seeing ways to improve them but refactoring that is beyond the scope of this PR.

File: nilearn/plotting/tests/test_displays.py [hunk 7]
```diff
@@ -271,7 +289,7 @@ def test_user_given_cmap_with_colorbar(img):
 
 @pytest.mark.parametrize("display", [OrthoSlicer, LYRZProjector])
 def test_data_complete_mask(display):
-    """This special case test is due to matplotlib 2.1.0.
+    """Test for a special case due to matplotlib 2.1.0.
```
Comment (ymzayek) [inline nilearn/plotting/tests/test_displays.py:292]: Maybe leave for another PR but maybe this patch and hence test are not needed anymore given that we support mpl>=3.3.0

File: pyproject.toml [hunk 8]
```diff
@@ -166,6 +166,9 @@ minversion = "6.0"
 addopts = "--doctest-modules -s -vv --durations=0"
 doctest_optionflags = "NORMALIZE_WHITESPACE ELLIPSIS"
 junit_family = "xunit2"
+pytest_plugins = [
+    "nilearn.plotting.tests.test_img_plotting._utils",
+]
```
Comment (ymzayek) [inline pyproject.toml:171]: Just for info, this is needed in both conftest and here?

**Review by ymzayek — APPROVED** [review]
Thx!

File: nilearn/plotting/tests/test_displays.py (hunk 7 above)
Comment (Remi-Gau) [reply nilearn/plotting/tests/test_displays.py:292]: let's flag this for another PR.

**Review by bthirion — APPROVED** [review]
LGTM, thx !

File: nilearn/plotting/displays/_slicers.py [hunk 9]
```diff
@@ -1,29 +1,28 @@
...
+from nilearn.plotting.find_cuts import find_cut_slices, find_xyz_cut_coords
 
 
 class BaseSlicer:
-    """BaseSlicer implementation which main purpose is to auto adjust
-    the axes size to the data with different layout of cuts. It create
+    """BaseSlicer implementation which main purpose is to auto adjust \
+    the axes size to the data with different layout of cuts.
+
+    It creates
```
Comment (NicolasGensollen) [inline nilearn/plotting/displays/_slicers.py:25]: Maybe merge the next line into this one ?

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @Remi-Gau !

**Comment by Remi-Gau** [comment]
TODO
- [ ] fix failing test on 3.7
- [ ] there seems to be some flaky tests: confirm and fix if needed

File: nilearn/plotting/displays/_slicers.py (hunk 9 above)
Comment (Remi-Gau) [reply nilearn/plotting/displays/_slicers.py]: done

File: nilearn/conftest.py
Comment (Remi-Gau) [reply nilearn/conftest.py:1]: @ymzayek 
are you reading this failure of the tests on python 3.7 without matplotlib the same way I am?

```
+ python -m pytest --pyargs nilearn --cov=nilearn
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.7.16/x64/lib/python3.7/site-packages/_pytest/config/__init__.py", line 774, in import_plugin
    __import__(importspec)
  File "/home/runner/work/nilearn/nilearn/nilearn/plotting/__init__.py", line 53, in <module>
    _set_mpl_backend()
  File "/home/runner/work/nilearn/nilearn/nilearn/plotting/__init__.py", line 14, in _set_mpl_backend
    import matplotlib
ModuleNotFoundError: No module named 'matplotlib'
```

see the rest here:

https://github.com/nilearn/nilearn/actions/runs/4787578339/jobs/8512995221?pr=3648#step:7:92

It seems that when trying to use the plugin fixtures that we have now moved to be specific to the plotting we now get an matplotlib import error.

**Comment by Remi-Gau** [comment]
There definitely is a flaky test:

```
FAILED nilearn/plotting/tests/test_matrix_plotting.py::test_sanitize_tri_error[None] - AssertionError: Regex pattern did not match.
 Regex: "Parameter tri needs to be one of {'full', 'lower', 'diag'}"
 Input: "Parameter tri needs to be one of {'lower', 'full', 'diag'}."
FAILED nilearn/plotting/tests/test_matrix_plotting.py::test_sanitize_tri_error[foo] - AssertionError: Regex pattern did not match.
 Regex: "Parameter tri needs to be one of {'full', 'lower', 'diag'}"
 Input: "Parameter tri needs to be one of {'lower', 'full', 'diag'}."
FAILED nilearn/plotting/tests/test_matrix_plotting.py::test_sanitize_tri_error[2] - AssertionError: Regex pattern did not match.
 Regex: "Parameter tri needs to be one of {'full', 'lower', 'diag'}"
 Input: "Parameter tri needs to be one of {'lower', 'full', 'diag'}."
```

File: nilearn/conftest.py
Comment (ymzayek) [reply nilearn/conftest.py:1]: Another thing you can try is to keep the name `testing_utils.py`. I'm wondering if it was named as such because of something specific to how pytest handles filenames. Just making guesses here. Feel free to transfer this discussion to a new issue

Comment (Remi-Gau) [reply nilearn/conftest.py:1]: > What was in `nilearn.plotting.tests.test_img_plotting._utils` exactly?

See here.

https://github.com/Remi-Gau/nilearn/blob/fd631396ab3134aa11a717339a2d0a9d1117c243/nilearn/plotting/tests/test_img_plotting/_utils.py

Comment (Remi-Gau) [reply nilearn/conftest.py:1]: > Another thing you can try is to keep the name `testing_utils.py`.

that did not work either

Comment (Remi-Gau) [reply nilearn/conftest.py:1]: Given the nature of the fixtures that were moved in conftest.py I think that they can be  renamed, refactored to be more reusable in other tests.

This will hopefully become more obvious as we keep refactoring the test suite in general.

**Comment by Remi-Gau** [comment]
This will be merged after the next release #3724
